### PR TITLE
feat(plasma-ui): Add swipe for whole sheet in `Sheet` component

### PR DESCRIPTION
**Pull request by neretin-trike**
Добавил возможность закрытия компонента `Sheet` по свайпу на всю высоту контента:

https://user-images.githubusercontent.com/26903236/231776474-f6c87b2c-9a61-4cac-8b1d-57d2e08291a8.mov

Также, если компонент имеет скролл, то закрыть его можно будет либо в самом начале скрола, либо по горизонтальной плашке:

https://user-images.githubusercontent.com/26903236/231776550-9673da53-fa54-4923-b416-fef719c0be9b.mov
 

:baby_chick: Download canary assets:

[colors--canary.482.4752482064.xml](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/colors--canary.482.4752482064.xml)  
[color_sberHealth_ios-swift--canary.482.4752482064.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_ios-swift--canary.482.4752482064.swift)  
[color_sberHealth_kotlin--canary.482.4752482064.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_kotlin--canary.482.4752482064.kt)  
[color_sberHealth_react-native--canary.482.4752482064.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_react-native--canary.482.4752482064.ts)  
[color_sbermarket_business_ios-swift--canary.482.4752482064.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_business_ios-swift--canary.482.4752482064.swift)  
[color_sbermarket_business_kotlin--canary.482.4752482064.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_business_kotlin--canary.482.4752482064.kt)  
[color_sbermarket_business_react-native--canary.482.4752482064.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_business_react-native--canary.482.4752482064.ts)  
[color_sbermarket_ios-swift--canary.482.4752482064.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_ios-swift--canary.482.4752482064.swift)  
[color_sbermarket_kotlin--canary.482.4752482064.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_kotlin--canary.482.4752482064.kt)  
[color_sbermarket_metro_ios-swift--canary.482.4752482064.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_metro_ios-swift--canary.482.4752482064.swift)  
[color_sbermarket_metro_kotlin--canary.482.4752482064.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_metro_kotlin--canary.482.4752482064.kt)  
[color_sbermarket_metro_react-native--canary.482.4752482064.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_metro_react-native--canary.482.4752482064.ts)  
[color_sbermarket_react-native--canary.482.4752482064.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_react-native--canary.482.4752482064.ts)  
[color_sbermarket_selgros_ios-swift--canary.482.4752482064.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_selgros_ios-swift--canary.482.4752482064.swift)  
[color_sbermarket_selgros_kotlin--canary.482.4752482064.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_selgros_kotlin--canary.482.4752482064.kt)  
[color_sbermarket_selgros_react-native--canary.482.4752482064.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_selgros_react-native--canary.482.4752482064.ts)  
[color_sbermarket_winelab_ios-swift--canary.482.4752482064.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_winelab_ios-swift--canary.482.4752482064.swift)  
[color_sbermarket_winelab_kotlin--canary.482.4752482064.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_winelab_kotlin--canary.482.4752482064.kt)  
[color_sbermarket_winelab_react-native--canary.482.4752482064.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_winelab_react-native--canary.482.4752482064.ts)  
[color_sberprime_ios-swift--canary.482.4752482064.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_ios-swift--canary.482.4752482064.swift)  
[color_sberprime_kotlin--canary.482.4752482064.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_kotlin--canary.482.4752482064.kt)  
[color_sberprime_react-native--canary.482.4752482064.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_react-native--canary.482.4752482064.ts)  
[PlasmaTokensColor--canary.482.4752482064.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/PlasmaTokensColor--canary.482.4752482064.swift)  
[shadow_sbermarket_react-native--canary.482.4752482064.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/shadow_sbermarket_react-native--canary.482.4752482064.ts)  
[typo_mage_ios-swift--canary.482.4752482064.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_ios-swift--canary.482.4752482064.swift)  
[typo_mage_kotlin--canary.482.4752482064.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_kotlin--canary.482.4752482064.kt)  
[typo_mage_react-native--canary.482.4752482064.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_react-native--canary.482.4752482064.ts)  
[typo_plasma_ios-swift--canary.482.4752482064.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_ios-swift--canary.482.4752482064.swift)  
[typo_plasma_kotlin--canary.482.4752482064.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_kotlin--canary.482.4752482064.kt)  
[typo_plasma_react-native--canary.482.4752482064.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_react-native--canary.482.4752482064.ts)  
[typo_ruler_ios-swift--canary.482.4752482064.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_ios-swift--canary.482.4752482064.swift)  
[typo_ruler_kotlin--canary.482.4752482064.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_kotlin--canary.482.4752482064.kt)  
[typo_ruler_react-native--canary.482.4752482064.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_react-native--canary.482.4752482064.ts)  
[typo_sage_ios-swift--canary.482.4752482064.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_ios-swift--canary.482.4752482064.swift)  
[typo_sage_kotlin--canary.482.4752482064.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_kotlin--canary.482.4752482064.kt)  
[typo_sage_react-native--canary.482.4752482064.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_react-native--canary.482.4752482064.ts)  
[typo_sbermarket_ios-swift--canary.482.4752482064.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_ios-swift--canary.482.4752482064.swift)  
[typo_sbermarket_kotlin--canary.482.4752482064.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_kotlin--canary.482.4752482064.kt)  
[typo_sbermarket_react-native--canary.482.4752482064.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_react-native--canary.482.4752482064.ts)  
[typo_soulmate_ios-swift--canary.482.4752482064.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_ios-swift--canary.482.4752482064.swift)  
[typo_soulmate_kotlin--canary.482.4752482064.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_kotlin--canary.482.4752482064.kt)  
[typo_soulmate_react-native--canary.482.4752482064.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_react-native--canary.482.4752482064.ts)

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @salutejs/plasma-temple@1.152.0-canary.482.4752482064.0
  npm install @salutejs/plasma-ui@1.184.0-canary.482.4752482064.0
  # or 
  yarn add @salutejs/plasma-temple@1.152.0-canary.482.4752482064.0
  yarn add @salutejs/plasma-ui@1.184.0-canary.482.4752482064.0
  ```
</details>

### CODE REVIEW

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-482/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-482/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-482/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-482/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-482/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-482/temple-storybook/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

File: packages/plasma-ui/src/components/Sheet/useSheetSwipe.ts [hunk 1]
```diff
@@ -1,56 +1,74 @@
...
     handleRef: React.RefObject<HTMLDivElement>;
     onClose: () => void;
 }) => {
     const { contentWrapperRef, handleRef, onClose } = args;
+    const [isTopScroll, setIsTopScroll] = useState(true);
+
+    useEffect(() => {
+        const contentWrapperEl = contentWrapperRef.current;
+        const handleEl = handleRef.current;
+        const contentEl = contentWrapperEl?.children[1];
```
Comment (Yakutoc) [inline packages/plasma-ui/src/components/Sheet/useSheetSwipe.ts:16]: [Уточнение]

@neretin-trike `children[1]` - по этому адресу всегда будет то что нам нужно? )

**Review by Yakutoc — APPROVED** [review]

File: packages/plasma-ui/src/components/Sheet/useSheetSwipe.ts (hunk 1 above)
Comment (neretin-trike) [reply packages/plasma-ui/src/components/Sheet/useSheetSwipe.ts:16]: ага, т.к. шит состоит из горизонтальной плашки [0] и контента [1]

**Review by Yeti-or — APPROVED** [review]
Проверил на iphone – все ок

File: packages/plasma-ui/src/components/Sheet/useSheetSwipe.ts (hunk 1 above)
Comment (neretin-trike) [reply packages/plasma-ui/src/components/Sheet/useSheetSwipe.ts:16]: ну по-идее оно всегда будет работать. Но да, лучше сделаю рефку, оно и по красивее будет выглядеть 👌

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-482/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-482/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-482/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-482/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-482/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-482/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-482/temple-storybook/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

**Pull request merged**